testcases/crypto/rsa_func.c

```diff
@@ -1963,6 +1963,164 @@ CK_RV do_VerifyRSA(struct PUBLISHED_TEST_SUITE_INFO * tsuite)
     return rc;
 }
 
+CK_RV do_RSAImplicitRejection(struct PUBLISHED_TEST_SUITE_INFO *tsuite)
+{
+    unsigned int i;
+    CK_BYTE decrypt[BIG_REQUEST];
+    CK_ULONG decrypt_len;
+    CK_MECHANISM mech;
+    CK_OBJECT_HANDLE priv_key = CK_INVALID_HANDLE;
+    CK_SLOT_ID slot_id = SLOT_ID;
+    CK_SESSION_HANDLE session;
+    CK_FLAGS flags;
+    CK_BYTE user_pin[PKCS11_MAX_PIN_LEN];
+    CK_ULONG user_pin_len;
+    CK_RV rc, loc_rc;
+
+    char *s;
+
+    // begin testsuite
+    testsuite_begin("%s Implicit Rejection.", tsuite->name);
+    testcase_rw_session();
+    testcase_user_login();
+
+    if (!is_ica_token(slot_id) && !is_soft_token(slot_id)) {
+        testsuite_skip(tsuite->tvcount,
+                       "Slot %u doesn't support Implicit Rejection",
+                       (unsigned int) slot_id);
+        goto testcase_cleanup;
+    }
+    // skip tests if the slot doesn't support this mechanism
+    if (!mech_supported(slot_id, tsuite->mech.mechanism)) {
+        testsuite_skip(tsuite->tvcount,
+                       "Slot %u doesn't support %s (0x%x)",
+                       (unsigned int) slot_id,
+                       mech_to_str(tsuite->mech.mechanism),
+                       (unsigned int) tsuite->mech.mechanism);
+        goto testcase_cleanup;
+    }
+
+    // iterate over test vectors
+    for (i = 0; i < tsuite->tvcount; i++) {
+
+        // get public exponent from test vector
+        if (p11_ahex_dump(&s, tsuite->tv[i].pub_exp,
+                          tsuite->tv[i].pubexp_len) == NULL) {
+            testcase_error("p11_ahex_dump() failed");
+            rc = -1;
+            goto testcase_cleanup;
+        }
+        // begin testcase
+        testcase_begin("%s Implicit Rejection with test vector %u."
+                       "\npubl_exp='%s', modbits=%lu, publ_exp_len=%lu.",
+                       tsuite->name, i, s,
+                       tsuite->tv[i].mod_len * 8,
+                       tsuite->tv[i].pubexp_len);
+
+        rc = CKR_OK;            // set rc
```

Comment:
so much space here before the comment?

Reply:
We have that same code at other places in this file as well. I just copied it as is.
Will remove the comment here anyway, its just telling the obvious anyway.